src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import core.basesyntax.suppliers.ColorSupplier;
+
+public abstract class Figure implements FigureInterface {
+    protected String color;
+
+    public Figure() {
+        color = ColorSupplier.DEFAULT_COLOR;
+    }
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    protected String getName() {
+        return getClass().getSimpleName().toLowerCase();
+    }
+
+    protected static String formatValue(double val) {
+        return String.format("%,.1f", val);
+    }
+
+    public abstract void draw();
```

Comment:
why do you repeat this?

Reply:
I have thought, that it's necessary. We need to specify that the method we implementing is abstract, or maybe change access modifier. Like so.